Tests fail due to log in github.com/ipfs/go-ds-flatfs after test has completed

There's a race condition between telling IPFS to shutdown [1], which stops the `checkpointLoop` the `go-ds-flatfs` is performing in the background, and the test ending. This causes issues as the `checkpointLoop` is regularly logging that it couldn't store final value of disk usage, for some reason.

We need to either be better at [shutting down the `fsrepo` we open](https://github.com/filecoin-project/bacalhau/blob/ff83fd6c66ab6364f0589d04930d3a9c7c8f12f0/pkg/ipfs/node.go#L317) or we need to get `go-ds-flatfs` to store the disk usage.

Example stacktrace:
```
panic: Log in goroutine after TestIPFSHostStorageSuite/TestIpfsApiCopyFile has completed: logger.go:159: 17:50:55.117 | ??? go-ds-flatfs@v0.5.1/flatfs.go:997 > ERROR flatfs could not store final value of disk usage to file, future estimates may be inaccurate

goroutine 487 [running]:
testing.(*common).logDepth(0xc001e7d520, {0xc002964000, 0xb7}, 0x3)
	/usr/local/go/src/testing/testing.go:894 +0x4c7
testing.(*common).log(...)
	/usr/local/go/src/testing/testing.go:876
testing.(*common).Logf(0xc001e7d520, {0x2f31da9?, 0xd0?}, {0xc0024f2e40?, 0xc00238b700?, 0x2e88880?})
	/usr/local/go/src/testing/testing.go:927 +0x5a
github.com/rs/zerolog.TestWriter.Write({{0x727eaf8?, 0xc001e7d520?}, 0x0?}, {0xc000bf01a0, 0x9b, 0xd0})
	/Users/distiller/go/pkg/mod/github.com/rs/zerolog@v1.28.0/writer.go:140 +0x364
bytes.(*Buffer).WriteTo(0xc0027b06c0, {0x35422c0?, 0xc00282c060?})
	/usr/local/go/src/bytes/buffer.go:252 +0x83
github.com/rs/zerolog.ConsoleWriter.Write({{0x35422c0, 0xc00282c060}, 0x1, {0x2f42bd7, 0xe}, {0xc001ca8180, 0x4, 0x4}, {0x0, 0x0, ...}, ...}, ...)
	/Users/distiller/go/pkg/mod/github.com/rs/zerolog@v1.28.0/console.go:145 +0x52a
github.com/rs/zerolog.levelWriterAdapter.WriteLevel(...)
	/Users/distiller/go/pkg/mod/github.com/rs/zerolog@v1.28.0/writer.go:25
github.com/rs/zerolog.(*Event).write(0xc001d302a0)
	/Users/distiller/go/pkg/mod/github.com/rs/zerolog@v1.28.0/event.go:78 +0x12a
github.com/rs/zerolog.(*Event).msg(0xc001d302a0, {0xc0004c0380, 0x63})
	/Users/distiller/go/pkg/mod/github.com/rs/zerolog@v1.28.0/event.go:149 +0x205
github.com/rs/zerolog.(*Event).Msg(...)
	/Users/distiller/go/pkg/mod/github.com/rs/zerolog@v1.28.0/event.go:108
github.com/filecoin-project/bacalhau/pkg/logger.(*zerologWriteSyncer).Write(0x2?, {0xc00292e000?, 0x63?, 0x49acd80?})
	/Users/distiller/repo/pkg/logger/logger.go:159 +0x73
go.uber.org/zap/zapcore.(*ioCore).Write(0xc00215a090, {0x2, {0xc0d5669fc6fcaa70, 0x764311d6, 0x49acd80}, {0x2f28468, 0x6}, {0x3112ecb, 0x55}, {0x1, ...}, ...}, ...)
	/Users/distiller/go/pkg/mod/go.uber.org/zap@v1.23.0/zapcore/core.go:99 +0xb5
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000db40d0, {0x0, 0x0, 0x0})
	/Users/distiller/go/pkg/mod/go.uber.org/zap@v1.23.0/zapcore/entry.go:255 +0x1d9
go.uber.org/zap.(*SugaredLogger).log(0xc0000104f8, 0x2, {0x0?, 0x0?}, {0xc0028d9f58?, 0x1?, 0x1?}, {0x0, 0x0, 0x0})
	/Users/distiller/go/pkg/mod/go.uber.org/zap@v1.23.0/sugar.go:287 +0xee
go.uber.org/zap.(*SugaredLogger).Error(...)
	/Users/distiller/go/pkg/mod/go.uber.org/zap@v1.23.0/sugar.go:134
github.com/ipfs/go-ds-flatfs.(*Datastore).checkpointLoop(0xc000c9e1e0)
	/Users/distiller/go/pkg/mod/github.com/ipfs/go-ds-flatfs@v0.5.1/flatfs.go:997 +0x25b
created by github.com/ipfs/go-ds-flatfs.Open
	/Users/distiller/go/pkg/mod/github.com/ipfs/go-ds-flatfs@v0.5.1/flatfs.go:291 +0x4a5
```

Examples of flakes:
https://app.circleci.com/pipelines/github/filecoin-project/bacalhau/2020/workflows/03bd3e5d-785c-4e14-9507-3fdd16c2ed30/jobs/13894
https://app.circleci.com/pipelines/github/filecoin-project/bacalhau/2062/workflows/006ed9ac-e607-4d39-9f26-bf53c78c2b41/jobs/14521

[1] Accomplished by either calling `Close` or by cancelling the context passed in

--- a/pkg/ipfs/node.go
+++ b/pkg/ipfs/node.go
@@ -11,6 +11,7 @@ import (
 	"sync"
 
 	"github.com/filecoin-project/bacalhau/pkg/system"
+	"github.com/hashicorp/go-multierror"
 	icore "github.com/ipfs/interface-go-ipfs-core"
 	ma "github.com/multiformats/go-multiaddr"
 	manet "github.com/multiformats/go-multiaddr/net"
@@ -25,6 +26,7 @@ import (
 	"github.com/ipfs/kubo/core/corehttp"
 	"github.com/ipfs/kubo/core/node/libp2p"
 	"github.com/ipfs/kubo/plugin/loader"
+	kuboRepo "github.com/ipfs/kubo/repo"
 	"github.com/ipfs/kubo/repo/fsrepo"
 	"github.com/libp2p/go-libp2p/core/peer"
 )
@@ -303,18 +305,29 @@ func createNode(ctx context.Context, cm *system.CleanupManager, cfg Config) (ico
 		return nil, nil, "", fmt.Errorf("failed to create repo dir: %w", err)
 	}
 
+	var repo kuboRepo.Repo
 	cm.RegisterCallback(func() error {
-		if err := os.RemoveAll(repoPath); err != nil { //nolint:govet
-			return fmt.Errorf("failed to clean up repo directory: %w", err)
+		var errs error
+		// We need to make sure we close the repo before we delete the disk contents as this will cause IPFS to print out messages about how
+		// 'flatfs could not store final value of disk usage to file', which is both annoying and can cause test flakes
+		// as the message can be written just after the test has finished but before the repo has been told by node
+		// that it's supposed to shut down.
+		if repo != nil {
+			if err := repo.Close(); err != nil { //nolint:govet
+				errs = multierror.Append(errs, fmt.Errorf("failed to close repo: %w", err))
+			}
 		}
-		return nil
+		if err := os.RemoveAll(repoPath); err != nil { //nolint:govet
+			errs = multierror.Append(errs, fmt.Errorf("failed to clean up repo directory: %w", err))
+		}
+		return errs
 	})
 
 	if err = createRepo(repoPath, cfg.getMode(), cfg.getKeypairSize()); err != nil {
 		return nil, nil, "", fmt.Errorf("failed to create repo: %w", err)
 	}
 
-	repo, err := fsrepo.Open(repoPath)
+	repo, err = fsrepo.Open(repoPath)
 	if err != nil {
 		return nil, nil, "", fmt.Errorf("failed to open temp repo: %w", err)
 	}

